blob/src/p2p.rs

```diff
@@ -112,7 +118,8 @@ where
 			sync_service,
 			gossip_cmd_receiver,
 		);
-		blob_handle.start_blob_cleaning_service(spawn_handle);
+		blob_handle.start_blob_cleaning_service(spawn_handle.clone());
```

Comment:
This is confusing. Instead of starting spawning stuff in the `new` function  make the `new` function return a BlobHandle and then call a method on the blobhandle to spawn any necessary resource.

`new` should be as atomic as possible as it is equal to constructors in other languages.

Reply:
You can see that this is an issue as `new` has dependenices on `pool` and `spawn_handle` yet in order to construct a BlobHandle you don't actually need it.

This is the same thing that we already have discussed. BlobHandle cannot be a dependency to itself.

Reply:
I still feel it's better this way since everything is located in the same place and we don't add too much in the service file. Feel free to change it, but i would prefer this way.